R/prepInputs.R

```diff
@@ -813,13 +808,66 @@ appendChecksumsTable <- function(checkSumFilePath, filesToChecksum,
   filesInArchive <- NULL
   if (!is.null(funWArgs$fun)) {
     if (file.exists(archive[1])) {
-      filesInArchive <- funWArgs$fun(archive[1], list = TRUE)
-      if ("Name" %in% names(filesInArchive)) {
-        # for zips, rm directories (length = 0)
-        filesInArchive <-
-          filesInArchive[filesInArchive$Length != 0,]$Name
-      } else { # untar
-        filesInArchive
+      if (!file_ext(archive[1]) == "rar") {
+        filesInArchive <- funWArgs$fun(archive[1], list = TRUE)
+        if ("Name" %in% names(filesInArchive)) {
+          # for zips, rm directories (length = 0)
+          filesInArchive <-
+            filesInArchive[filesInArchive$Length != 0,]$Name
+        } else { # untar
+          filesInArchive
+        }
+      } else {
+        message(paste0("The archive is a .rar file. preProcess will try a system call of 'unrar'."))
+        wd <- getwd()
+        tempDir <- file.path(dirname(archive[1]), "extractedFiles")
+        dir.create(tempDir, showWarnings = FALSE)
+        setwd(tempDir)
+        OSystem <- Sys.info()["sysname"]
+        if (OSystem == "Windows") {
+          find7zPath <- list.files("C:/Program Files/7-Zip", pattern = "7z.exe", recursive = TRUE, full.names = TRUE)
```

Comment:
this may not be the install location, esp. on 64 bit systems (which are fairly standard now).

you should use `Sys.which()` to find the executable.

Reply:
This didn't work for the Windows machine...

Reply:
`Sys.which()` will tell you if the executable is installed on one of the usual places -- it came back with an empty string so it can't find it on your system. Maybe a PATH issue? Not sure, because `C:/Program Files/` should be on the PATH. I can't test this easily since I don't have access to Windows.

Reply:
maybe also try `Sys.which2()` from [here](https://stackoverflow.com/a/34031214)

Reply:
also, what does `where 7z.exe` return on Windows from cmd line?

Reply:
try `Sys.which` and `where` without the file extension too

Reply:
if you use `Sys.which2()` be sure to credit the orig source: https://stat.ethz.ch/pipermail/r-devel/2015-May/071199.html

Reply:
It doesn't find it. I saw the Sys.which2() and also tested it, but it also doesn't work... >< That's why I came up with this slow swe system